ctapipe/io/hdf5eventsource.py

```diff
@@ -273,6 +277,13 @@ def has_simulated_dl1(self):
                 return True
         return False
 
+    @property
```

Comment:
could be lazy?

Reply:
Lazy as in @lazyproperty? I don't really see where this should create problems. Besides that, should we then change this for `has_simulated_dl1` and `is_simulation` too?